src/ClientAssetsManager.php

```diff
@@ -203,13 +207,13 @@ private function initJavaScriptOptimization()
             }
             $cachePath = $this->cacheDir . '/' . sha1(implode('|', $hash)) . '.js';
             if (!is_file($cachePath)) {
-                // Combined script is not yet available - generate it and put into cache
-                $content = [];
+                // Combined script is not yet available - generate, minified it and put into cache
+                $content = '';
                 foreach ($parts as $part) {
-                    $content[] = '/* [' . basename($part) . '] */;';
-                    $content[] = file_get_contents($part);
+                    $content .= file_get_contents($part);
                 }
-                file_put_contents($cachePath, implode("\n", $content));
+                $minifiedJs = Minifier::minify($content, ['flaggedComments' => false]);
```

Comment:
I don't think that any kind of minifier that is not based on the AST is safe to use by default. Language and its syntax evolve so it is expected to fail in its assumptions.